src/main/java/com/softserveinc/dokazovi/service/impl/UserServiceImpl.java

```diff
@@ -184,12 +193,12 @@ public Page<UserDTO> findRandomExpertPreview(Set<Integer> directionsIds, Pageabl
      * @param user user received from Auth controller
      */
     @Override
-    public void setEnableTrue(UserEntity user) {
+    public void changeEnable(UserEntity user) {
         UserEntity userEntity = userRepository.findById(user.getId()).orElse(null);
         if (userEntity == null) {
             throw new BadRequestException("Something went wrong!!!");
         }
-        userEntity.setEnabled(true);
+        userEntity.setEnabled(!userEntity.getEnabled());
```

Comment:
This is a bad idea because the action is not idempotent - that is, calling `changeEnable` again when user is already enabled would cause it being disabled, and it is not always easy to track such pieces of code.
Somebody after you will just blindly call this method without understanding it fully, and will potentially break the app.

If you absolutely need to have the "switch" behavior, it is better to do it at the frontend side.
Here, I would suggest having this:

```java
@Override
public void setEnabled(UserEntity user, boolean isEnabled) {
  // ...
  userEntity.setEnabled(isEnabled);
  // ...
}
```

And to ensure that you can fetch `enabled` status from frontend side to then ask to set it to the needed value.